lightning/src/ln/channel.rs

```diff
@@ -2132,110 +2132,97 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
+								update_fulfill_htlcs.push(update_fulfill_msg_option.unwrap());
+								if let Some(mut additional_monitor_update) = additional_monitor_update_opt {
+									monitor_update.updates.append(&mut additional_monitor_update.updates);
+								}
+							},
+							Err(e) => {
+								if let ChannelError::Ignore(_) = e {}
+								else {
+									panic!("Got a non-IgnoreError action trying to fulfill holding cell HTLC");
 								}
 							}
-						},
-						&HTLCUpdateAwaitingACK::FailHTLC { htlc_id, ref err_packet } => {
-							match self.get_update_fail_htlc(htlc_id, err_packet.clone()) {
-								Ok(update_fail_msg_option) => update_fail_htlcs.push(update_fail_msg_option.unwrap()),
-								Err(e) => {
-									if let ChannelError::Ignore(_) = e {}
-									else {
-										panic!("Got a non-IgnoreError action trying to fail holding cell HTLC");
-									}
+						}
+					},
+					&HTLCUpdateAwaitingACK::FailHTLC { htlc_id, ref err_packet } => {
+						match self.get_update_fail_htlc(htlc_id, err_packet.clone()) {
+							Ok(update_fail_msg_option) => update_fail_htlcs.push(update_fail_msg_option.unwrap()),
+							Err(e) => {
+								if let ChannelError::Ignore(_) = e {}
+								else {
+									panic!("Got a non-IgnoreError action trying to fail holding cell HTLC");
 								}
 							}
-						},
-					}
-					if err.is_some() {
-						self.holding_cell_htlc_updates.push(htlc_update);
-						if let Some(ChannelError::Ignore(_)) = err {
-							// If we failed to add the HTLC, but got an Ignore error, we should
-							// still send the new commitment_signed, so reset the err to None.
-							err = None;
 						}
-					}
+					},
 				}
 			}
-			//TODO: Need to examine the type of err - if it's a fee issue or similar we may want to
-			//fail it back the route, if it's a temporary issue we can ignore it...
-			match err {
-				None => {
-					if update_add_htlcs.is_empty() && update_fulfill_htlcs.is_empty() && update_fail_htlcs.is_empty() && self.holding_cell_update_fee.is_none() {
-						// This should never actually happen and indicates we got some Errs back
-						// from update_fulfill_htlc/update_fail_htlc, but we handle it anyway in
-						// case there is some strange way to hit duplicate HTLC removes.
-						return Ok(None);
-					}
-					let update_fee = if let Some(feerate) = self.holding_cell_update_fee {
-							self.pending_update_fee = self.holding_cell_update_fee.take();
-							Some(msgs::UpdateFee {
-								channel_id: self.channel_id,
-								feerate_per_kw: feerate as u32,
-							})
-						} else {
-							None
-						};
+			if update_add_htlcs.is_empty() && update_fulfill_htlcs.is_empty() && update_fail_htlcs.is_empty() && self.holding_cell_update_fee.is_none() {
+				// Hitting this case indicates that we got some Errs back from update_fulfill_htlc
+				// or update_fail_htlc.
+				log_warn!(logger, "Attempted to fulfill or fail an HTLC that was already removed");
```

Comment:
Hmm, wait, this case also happens if we hit the new update_add failures, so we shouldn't really print a scary warning here, it may just be a fee race condition. Also comment seems wrong?